### PR TITLE
Refactor DataTable Resizer.js using hooks

### PR DESCRIPTION
**Pull request by laka3000**
#### What does this PR do?
Refactor `src/js/components/DataTable/Resizer.js` using hooks

#### Where should the reviewer start?
`src/js/components/DataTable/Resizer.js` 

#### What testing has been done on this PR?
Running `yarn storybook` and checking its behavior, it's used in `/story/datatable--tunable`

#### How should this be manually tested?
Running `yarn storybook` and checking its behavior, it's used in `/story/datatable--tunable`

#### Any background context you want to provide?
No

#### What are the relevant issues?
https://github.com/grommet/grommet/issues/3394

#### Screenshots (if appropriate)
--

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/Resizer.js [hunk 1]
```diff
@@ -12,55 +12,61 @@ const ResizerBox = styled(Box)`
...
+    onResize(property)(nextWidth);
+  };
 
-  onMouseDown = event => {
-    if (this.ref.current) {
-      let element = this.ref.current;
+  const onMouseUp = () => {
+    document.removeEventListener('mouseup', onMouseUp);
+    document.removeEventListener('mousemove', onMouseMove);
+    setState({ start: undefined, width: undefined });
```
Comment (negreirosleo) [inline src/js/components/DataTable/Resizer.js]: setState({ ...state, start: undefined, width: undefined });

Comment (laka3000) [reply src/js/components/DataTable/Resizer.js]: Thanks, fixed :heart:

File: src/js/components/DataTable/Resizer.js [hunk 2]
```diff
@@ -12,55 +12,61 @@ const ResizerBox = styled(Box)`
...
-
-  onMouseUp = () => {
-    document.removeEventListener('mouseup', this.onMouseUp);
-    document.removeEventListener('mousemove', this.onMouseMove);
-    this.setState({ start: undefined, width: undefined });
-  };
+  useEffect(() => {
+    if (state.event === 'onMouseDown') {
+      document.addEventListener('mousemove', onMouseMove);
+      document.addEventListener('mouseup', onMouseUp);
```
Comment (negreirosleo) [inline src/js/components/DataTable/Resizer.js]: Its a good habit to pass the dependencies to the useEffect array,
```js
[state,onMouseMove,onMouseUp]
```
also, its a good idea to use useCallback with onMouseMove and onMouseUp

Comment (laka3000) [reply src/js/components/DataTable/Resizer.js]: Thank you so much, fixed!

File: src/js/components/DataTable/Resizer.js [hunk 3]
```diff
@@ -12,55 +12,68 @@ const ResizerBox = styled(Box)`
...
 `;
 
-class Resizer extends Component {
-  state = {};
+const Resizer = ({ onResize, property, theme }) => {
+  const [state , setState] = useState({
+    start: undefined,
+    width: undefined,
+    event: undefined,
+  });
```
Comment (ShimiSun) [inline src/js/components/DataTable/Resizer.js]: Can we please separate those from one state to multiple different states?

Comment (laka3000) [reply src/js/components/DataTable/Resizer.js]: Fixed!

**Comment by ShimiSun** [comment]
Thank you for the contribution @laka3000 !

**Pull request merged**